app/forms/idv/profile_form.rb

```diff
@@ -38,5 +38,27 @@ def ssn_is_unique
         errors.add :ssn, I18n.t('idv.errors.duplicate_ssn')
       end
     end
+
+    def dob_is_sane
+      return unless dob_date?
+      return unless dob_in_the_past?
+    end
+
+    def dob_in_the_past?
+      date = Date.parse(dob)
+      if date > Date.today
+        errors.add :dob, I18n.t('idv.errors.dob_in_future')
+        false
+      else
+        true
+      end
+    end
+
+    def dob_date?
+      Date.parse(dob)
+    rescue
+      errors.add :dob, I18n.t('idv.errors.bad_dob')
+      false
+    end
```

Comment:
I have refactored to have only one DOB error message and parse the date string only once.